src/basic/socket-util.h

```diff
@@ -147,6 +148,16 @@ int flush_accept(int fd);
 
 struct cmsghdr* cmsg_find(struct msghdr *mh, int level, int type, socklen_t length);
 
+#define SOCKADDR_LL_LEN(sa)                                             \
+        ({                                                              \
+                const struct sockaddr_ll *_sa = &(sa);                  \
+                unsigned long _mac_len = sizeof(_sa->sll_addr);         \
+                assert(_sa->sll_family == AF_PACKET);                   \
+                if (be16toh(_sa->sll_hatype) == ARPHRD_INFINIBAND)      \
+                        _mac_len = INFINIBAND_ALEN;                     \
```

Comment:
I'd feel better if this would be written as:

```c
if (be16toh(_sa->sll_hatype) == ARPHRD_INFINIBAND))
        _mac_len = MAX(_mac_len, INFINIBAND_ALEN);
```

i.e. even though we know that infiniband's ALEN is larger than sizeof(_sa->sll_addr), let the compiler figure that out... It will optimize this away anyway... In fact, I'd even add a similar line for ARPHDR_ETHERNET too, where it is fully redundant, just to make it clear to the reader what is going on, without expecting him to know the size implications here...